src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,268 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.6
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\View\Helper;
+
+use Cake\View\Helper;
+use Cake\View\StringTemplateTrait;
+use LogicException;
+
+/**
+ * BreadcrumbsHelper to register and display a breadcrumb trail for your views
+ *
+ * @property \Cake\View\Helper\UrlHelper $Url
+ */
+class BreadcrumbsHelper extends Helper
+{
+
+    use StringTemplateTrait;
+
+    /**
+     * Other helpers used by BreadcrumbsHelper
+     *
+     * @var array
+     */
+    public $helpers = ['Url'];
+
+    /**
+     * Default config for the helper.
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'templates' => [
+            'wrapper' => '<ul{{attrs}}>{{content}}</ul>',
+            'item' => '<li{{attrs}}><a href="{{link}}"{{innerAttrs}}>{{title}}</a></li>{{separator}}',
+            'itemWithoutLink' => '<li{{attrs}}><span{{innerAttrs}}>{{title}}</span></li>{{separator}}',
+            'separator' => '<li{{attrs}}><span{{innerAttrs}}>{{separator}}</span></li>'
```

Comment:
It would be better to have separator as empty string by default IMO. Separators are generally handled with CSS. That's what popular frameworks like Bootstrap and Foundation do.